utils/model_validation_cache.py

```
@@ -0,0 +1,479 @@
+
+This module provides intelligent caching for model validation, provider resolution,
+and capability checks to reduce repeated validation computations.
+"""
+
+import hashlib
+import json
+import logging
+import threading
+import time
+from collections import OrderedDict
+from typing import Any, Dict, List, Optional
```

Ruff (0.11.9)

`typing.Dict` is deprecated, use `dict` instead

(UP035)